lib/matplotlib/backends/_backend_tk.py

```diff
@@ -454,15 +455,13 @@ def destroy(*args):
             self._shown = True
 
     def destroy(self, *args):
-        if self.window is not None:
-            #self.toolbar.destroy()
-            if self.canvas._idle_callback:
-                self.canvas._tkcanvas.after_cancel(self.canvas._idle_callback)
-            self.window.destroy()
-        if Gcf.get_num_fig_managers() == 0:
-            if self.window is not None:
-                self.window.quit()
-        self.window = None
+        if self.canvas._idle_callback:
+            self.canvas._tkcanvas.after_cancel(self.canvas._idle_callback)
+
+        self.window.destroy()
```

Comment:
We used to drop the window, presumably to avoid calling `destroy()` on it twice (?).  Is that a concern we should have?

Reply:
Calling destroy twice was an artifact of mistakenly binding <destroy> instead of using protocol. It should be fine now.